compositor_pipeline/src/pipeline/input/rtp/depayloader.rs

```diff
@@ -142,21 +164,103 @@ impl AudioDepayloader {
         packet: rtp::packet::Packet,
     ) -> Result<Option<EncodedChunk>, DepayloadingError> {
         match self {
-            AudioDepayloader::Opus(depayloader) => {
+            AudioDepayloader::Opus {
+                depayloader,
+                previous_timestamp,
+                rollover_count,
+            } => {
                 let kind = EncodedChunkKind::Audio(AudioCodec::Opus);
                 let opus_packet = depayloader.depacketize(&packet.payload)?;
 
                 if opus_packet.is_empty() {
                     return Ok(None);
                 }
 
+                let timestamp = timestamp_with_rollover(
+                    rollover_count,
+                    *previous_timestamp,
+                    packet.header.timestamp,
+                );
+
+                *previous_timestamp = packet.header.timestamp;
+
                 Ok(Some(EncodedChunk {
                     data: opus_packet,
-                    pts: Duration::from_secs_f64(packet.header.timestamp as f64 / 48000.0),
+                    pts: Duration::from_secs_f64(timestamp as f64 / 48000.0),
                     dts: None,
                     kind,
                 }))
             }
         }
     }
 }
+
+fn timestamp_with_rollover(
+    rollover_count: &mut usize,
+    previous_timestamp: u32,
+    current_timestamp: u32,
+) -> u64 {
+    let timestamp_diff = u32::abs_diff(previous_timestamp, current_timestamp);
+    if timestamp_diff >= u32::MAX / 2 {
+        if previous_timestamp > current_timestamp {
+            *rollover_count += 1;
+        } else {
+            // We received a packet from before the rollover, so we need to decrement the count
+            *rollover_count = rollover_count.saturating_sub(1);
+        }
+    }
+
+    (*rollover_count as u64) * (u32::MAX as u64 + 1) + current_timestamp as u64
+}
```

Comment:
I think this won't work. If the stream starts at random PTS (as in RFC), this `rollover_count * range` is not the correct timestamp.

Assume rollover happens at 10s. If I understand correctly, if the stream starts at timestamp 9s, after rolling over, let's say after 2s, its timestamp will be set to `10s + 1s` (current timestamp is 1s since 9s -[1s]-> 10s/0s -[1s]-> 1s) but should be set to 2s.

Correct me if I missunderstod something here.